### PR TITLE
Simplify validation

### PR DESCRIPTION
**Pull request by elcorbs**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/PA-399
### *What is the problem we're trying to solve*

The current controller/ validation logic is quite confusing to get your head around when joining the project. 

For example, the search address validation was occurring in 3 places.
### *What changes have we introduced*

This tries to simplify the validation logic by just using one validation class in each usecases.
In the case of the search validation, there is still validation of the model state separately to the main validation, the plan is to try and move this at a later date to stop this PR being too big.

I've also tried to simplify the response objects by removing parts that weren't being used and trying the flatten some of the response structure.
I've also introduced separate response objects for successful and unsuccessful responses.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: AddressesAPI.Tests/V1/SearchAddressRequestTests.cs [hunk 1]
```diff
@@ -8,26 +8,35 @@ namespace AddressesAPI.Tests.V1
     [TestFixture]
     public class SearchAddressRequestTests
     {
+        private static string _bothGazetteers;
+        private static string _localGazetteer;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _bothGazetteers = GlobalConstants.Gazetteer.Both.ToString();
```
Comment (Renny4Real) [inline AddressesAPI.Tests/V1/SearchAddressRequestTests.cs:17]: This reads nicely! I sometimes get confused when I see `GlobalConstants.Gazetteer.Both`.

File: AddressesAPI/V1/Boundary/Responses/Metadata/APIError.cs [hunk 2]
```diff
@@ -1,37 +1,10 @@
...
-        {
-            Errors = new List<ExecutionError> { new ExecutionError(ex) };
-        }
-
-        public APIError(ExecutionError error)
-        {
-            Errors = new List<ExecutionError> { error };
-        }
-
-
```
Comment (Renny4Real) [inline AddressesAPI/V1/Boundary/Responses/Metadata/APIError.cs:35]: Nice 🎉

File: AddressesAPI/V1/Boundary/Requests/SearchAddressRequest.cs [hunk 3]
```diff
@@ -14,12 +8,12 @@ namespace AddressesAPI.V1.Boundary.Requests
...
     public class SearchAddressRequest
     {
-        private GlobalConstants.Gazetteer _gazetteer;
+        private string _gazetteer;
 
         public SearchAddressRequest()
         {
             AddressStatus = "approved preferred";
-            Gazetteer = GlobalConstants.Gazetteer.Both;
+            Gazetteer = GlobalConstants.Gazetteer.Both.ToString();
```
Comment (Renny4Real) [inline AddressesAPI/V1/Boundary/Requests/SearchAddressRequest.cs:16]: Would we remove this, perhaps in a later PR since we have validation for these?

**Review by Renny4Real — APPROVED** [review]
Looks good 🎉

File: AddressesAPI/V1/Boundary/Requests/SearchAddressRequest.cs (hunk 3 above)
Comment (elcorbs) [reply AddressesAPI/V1/Boundary/Requests/SearchAddressRequest.cs:16]: Yeah maybe, I think it's there to set the default value at the moment.

**Pull request merged**